app/src/main/java/org/mozilla/rocket/privately/browse/BrowserFragment.kt

```diff
@@ -381,18 +388,29 @@ class BrowserFragment : LocaleAwareFragment(),
 
     private fun initTrackerView(parentView: View) {
         trackerPopup = TrackerPopup(parentView.context)
-
+        trackerPopup.setSwitchToggled(isTurboModeEnabled(requireContext()))
         trackerPopup.onSwitchToggled = { enabled ->
-            val appContext = (parentView.context.applicationContext as FocusApplication)
-            appContext.settings.privateBrowsingSettings.setTurboMode(enabled)
-            sessionManager.focusSession?.engineSession?.tabView?.setContentBlockingEnabled(enabled)
-
+            app().settings.privateBrowsingSettings.setTurboMode(enabled)
+            setTrackingProtectionEnabled(enabled)
+            // TODO: move to Session.Observer.onTrackerBlockingEnabledChanged
+            // for now the callback has a bug in version 0.52.0
             bottomBarItemAdapter.setTrackerSwitch(enabled)
             stop()
             reload()
         }
     }
 
+    private fun setTrackingProtectionEnabled(enabled: Boolean) {
+        if (enabled) {
+            EngineSession.TrackingProtectionPolicy.all()
+        } else {
```

Comment:
I'm not sure why you need this else block :P

Reply:
we also need to set trackingProtectionPolicy as null when tracking protection is disabled